src/server/editorServices.ts

```diff
@@ -1372,24 +1372,39 @@ namespace ts.server {
             this.projectToSizeMap.forEach(val => (availableSpace -= (val || 0)));
 
             let totalNonTsFileSize = 0;
+
             for (const f of fileNames) {
                 const fileName = propertyReader.getFileName(f);
                 if (hasTypeScriptFileExtension(fileName)) {
                     continue;
                 }
+
                 totalNonTsFileSize += this.host.getFileSize(fileName);
+
                 if (totalNonTsFileSize > maxProgramSizeForNonTsFiles) {
+                    logExceedLimit.call(this, totalNonTsFileSize);
                     // Keep the size as zero since it's disabled
                     return true;
                 }
             }
 
             if (totalNonTsFileSize > availableSpace) {
+                logExceedLimit.call(this, totalNonTsFileSize);
                 return true;
             }
 
             this.projectToSizeMap.set(name, totalNonTsFileSize);
             return false;
+
+            function logExceedLimit(this: ProjectService, totalNonTsFileSize: number) {

```

Comment:
why do you need the `this` here?

Reply:
just pass in the logger/host, or make the function return a string.

Reply:
Sure, that would be even better.